main_code/Email/inbox-dialogflow.py

```diff
@@ -0,0 +1,373 @@
+'''
+Two credential files required. For dialogflow & Gmail
+
+References:
+Getting started with Authentication (GCP)
+https://cloud.google.com/docs/authentication/getting-started#auth-cloud-implicit-python
+
+Set path
+https://stackoverflow.com/questions/45501082/set-google-application-credentials-in-python-project-to-use-google-api
+
+Gmail API SCOPES https://developers.google.com/gmail/api/auth/scopes
+'''
+
+from __future__ import print_function
+import pickle
+import os.path
+from googleapiclient.discovery import build
+from google_auth_oauthlib.flow import InstalledAppFlow
+from google.auth.transport.requests import Request
+
+
+import base64
+import email
+import re
+from apiclient import errors
+from email.mime.text import MIMEText
+
+import win32clipboard
+import dialogflow
+import os
+import sys
+sys.path.append('../automations')
+
+
+# If modifying these scopes, delete the file token.pickle.
+SCOPES = ['https://mail.google.com/'] # full access to the email account
+
+
+def test_automations():
+    '''This function is for testing automations. No input; no return.
+
+    Choose task string from exmaple_task. This represents a mock response from dialogflow
+    The value from the spreadsheet is placed in the clipboard
+
+    PartNameFromCompany   Given the company name, return the part name their provided    auto_get_part_name_2.py
+    GetContactEmail       Given the partID, return the email of the supplier    auto_get_email.py
+    PartNameFromId        Given the partID, return the part name    auto_get_part_name_1.py
+    PartIdFromCompany     Given Company's name, return the partID they provide    auto_get_partID.py
+
+    '''
+    example_tasks = ["PartNameFromCompany A++","GetContactEmail GERTB125", "PartNameFromId 0988FHG2", "PartIdFromCompany RocketScience"]
+    expected_values_from_spreadsheet = ["Haltegriff Pl", "wils1170@msu.edu", "Gear", "GERTB125"]
+
+    action = get_action(example_tasks[3])
+    print(action)
+    take_action(action)
+
+
+def ModifyMessage(service, user_id, msg_id, msg_labels):
+  """Modify the Labels on the given Message.
+
+  Args:
+    service: Authorized Gmail API service instance.
+    user_id: User's email address. The special value "me"
+    can be used to indicate the authenticated user.
+    msg_id: The id of the message required.
+    msg_labels: The change in labels.
+
+  Returns:
+    Modified message, containing updated labelIds, id and threadId.
+  """
+  try:
+    message = service.users().messages().modify(userId=user_id, id=msg_id,
+                                                body=msg_labels).execute()
+
+    label_ids = message['labelIds']
+
+    print( 'Message ID: %s - With Label IDs %s' % (msg_id, label_ids))
+    return message
+  except errors.HttpError as error:

```

Comment:
Are you sure you don't want to raise an exception here instead of printing out the error and continuing?